apl-core/src/main/java/com/apollocurrency/aplwallet/apl/alias/service/AliasServiceImpl.java

```diff
@@ -0,0 +1,184 @@
+            new DbClause.LikeClause("alias_name_lower", aliasName.toLowerCase()), from, to
+        );
+
+        return converter.convert(aliasIterator);
+    }
+
+    @Override
+    public Alias getAliasById(long id) {
+        return aliasTable.getAlias(id);
+    }
+
+    @Override
+    public void deleteAlias(final String aliasName) {
+        final Alias alias = getAliasByName(aliasName);
+        final AliasOffer offer = offerTable.getOffer(alias);
+        final int height = blockchain.getHeight();
+        if (offer != null) {
+            offer.setPriceATM(Long.MAX_VALUE);
+            offer.setHeight(height);
+            offerTable.deleteAtHeight(offer, height);
+        }
+        alias.setHeight(height);
+        aliasTable.deleteAtHeight(alias, height);
+    }
+
+    @Override
+    public void addOrUpdateAlias(Transaction transaction, MessagingAliasAssignment attachment) {
+        Alias alias = getAliasByName(attachment.getAliasName());
+        if (alias == null) {
+            alias = new Alias(transaction, attachment, blockchain.getHeight(), blockchain.getLastBlockTimestamp());
+        } else {
+            alias.setHeight(blockchain.getHeight());
+            alias.setAccountId(transaction.getSenderId());
+            alias.setAliasURI(attachment.getAliasURI());
+            alias.setTimestamp(blockchain.getLastBlockTimestamp());
+        }
+        aliasTable.insert(alias);
+    }
+
+    @Override
+    public void sellAlias(Transaction transaction, MessagingAliasSell attachment) {
+        final String aliasName = attachment.getAliasName();
+        final long priceATM = attachment.getPriceATM();
+        final long buyerId = transaction.getRecipientId();
+        if (priceATM > 0) {
+            Alias alias = getAliasByName(aliasName);
+            AliasOffer offer = offerTable.getOffer(alias);
+            if (offer == null) {
+                offerTable.insert(new AliasOffer(alias.getId(), priceATM, buyerId, blockchain.getHeight()));
+            } else {
+                offer.setHeight(blockchain.getHeight());
+                offer.setPriceATM(priceATM);
+                offer.setBuyerId(buyerId);
+                offerTable.insert(offer);
+            }
+        } else {
+            changeOwner(buyerId, aliasName);
+        }
+    }
+
+    @Override
+    public void changeOwner(long newOwnerId, String aliasName) {
+        Alias alias = getAliasByName(aliasName);
+        alias.setHeight(blockchain.getHeight());
+        alias.setAccountId(newOwnerId);
+        alias.setTimestamp(blockchain.getLastBlockTimestamp());
+        aliasTable.insert(alias);
+        AliasOffer offer = offerTable.getOffer(alias);
+        if (offer != null) {
+            offer.setHeight(blockchain.getHeight());
+            offer.setPriceATM(Long.MAX_VALUE);
+            offerTable.deleteAtHeight(offer, blockchain.getHeight());
+        }
+    }
+
+    @Override
+    public AliasOffer getOffer(Alias alias) {
```

Comment:
It would be good to add check input to prevent NPE - Objects.requireNonNull(..)

Reply:
done